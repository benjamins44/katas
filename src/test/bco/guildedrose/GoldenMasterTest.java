package bco.guildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

class GoldenMasterTest {

  @Test
  void goldenMaster() {
    final StringBuffer result = new StringBuffer();
    result.append("OMGHAI!");

    final Product[] items = new Product[]{
        new Dexterity(10, 20), //
        new AgedBrie(2, 0), //
        new Elixir(5, 7), //
        new Sulfura(0, 80), //
        new Sulfura(-1, 80),
        new BackstagePasses(15, 20),
        new BackstagePasses(10, 49),
        new BackstagePasses(5, 49),
        // this conjured item does not work properly yet
        new ConjuredManaCake(3, 6)};

    final GildedRose app = new GildedRose(items);

    final int days = 10;

    for (int i = 0; i < days; i++) {
      result.append("-------- day ").append(i).append(" --------");
      result.append("name, sellIn, quality");
      for (final Product item : items) {
        result.append(item);
      }
      app.updateQuality();
    }

    Approvals.verify(result);
  }
}
