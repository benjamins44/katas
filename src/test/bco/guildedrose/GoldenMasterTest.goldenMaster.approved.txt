OMGHAI!-------- day 0 --------name, sellIn, quality+5 Dexterity Vest, 10, 20Aged Brie, 2, 0Elixir of the Mongoose, 5, 7Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 15, 20Backstage passes to a TAFKAL80ETC concert, 10, 49Backstage passes to a TAFKAL80ETC concert, 5, 49Conjured Mana Cake, 3, 6-------- day 1 --------name, sellIn, quality+5 Dexterity Vest, 9, 19Aged Brie, 1, 1Elixir of the Mongoose, 4, 6Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 14, 21Backstage passes to a TAFKAL80ETC concert, 9, 50Backstage passes to a TAFKAL80ETC concert, 4, 50Conjured Mana Cake, 2, 4-------- day 2 --------name, sellIn, quality+5 Dexterity Vest, 8, 18Aged Brie, 0, 2Elixir of the Mongoose, 3, 5Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 13, 22Backstage passes to a TAFKAL80ETC concert, 8, 50Backstage passes to a TAFKAL80ETC concert, 3, 50Conjured Mana Cake, 1, 2-------- day 3 --------name, sellIn, quality+5 Dexterity Vest, 7, 17Aged Brie, -1, 4Elixir of the Mongoose, 2, 4Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 12, 23Backstage passes to a TAFKAL80ETC concert, 7, 50Backstage passes to a TAFKAL80ETC concert, 2, 50Conjured Mana Cake, 0, 0-------- day 4 --------name, sellIn, quality+5 Dexterity Vest, 6, 16Aged Brie, -2, 6Elixir of the Mongoose, 1, 3Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 11, 24Backstage passes to a TAFKAL80ETC concert, 6, 50Backstage passes to a TAFKAL80ETC concert, 1, 50Conjured Mana Cake, -1, 0-------- day 5 --------name, sellIn, quality+5 Dexterity Vest, 5, 15Aged Brie, -3, 8Elixir of the Mongoose, 0, 2Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 10, 25Backstage passes to a TAFKAL80ETC concert, 5, 50Backstage passes to a TAFKAL80ETC concert, 0, 50Conjured Mana Cake, -2, 0-------- day 6 --------name, sellIn, quality+5 Dexterity Vest, 4, 14Aged Brie, -4, 10Elixir of the Mongoose, -1, 0Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 9, 27Backstage passes to a TAFKAL80ETC concert, 4, 50Backstage passes to a TAFKAL80ETC concert, -1, 0Conjured Mana Cake, -3, 0-------- day 7 --------name, sellIn, quality+5 Dexterity Vest, 3, 13Aged Brie, -5, 12Elixir of the Mongoose, -2, 0Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 8, 29Backstage passes to a TAFKAL80ETC concert, 3, 50Backstage passes to a TAFKAL80ETC concert, -2, 0Conjured Mana Cake, -4, 0-------- day 8 --------name, sellIn, quality+5 Dexterity Vest, 2, 12Aged Brie, -6, 14Elixir of the Mongoose, -3, 0Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 7, 31Backstage passes to a TAFKAL80ETC concert, 2, 50Backstage passes to a TAFKAL80ETC concert, -3, 0Conjured Mana Cake, -5, 0-------- day 9 --------name, sellIn, quality+5 Dexterity Vest, 1, 11Aged Brie, -7, 16Elixir of the Mongoose, -4, 0Sulfuras, Hand of Ragnaros, 0, 80Sulfuras, Hand of Ragnaros, -1, 80Backstage passes to a TAFKAL80ETC concert, 6, 33Backstage passes to a TAFKAL80ETC concert, 1, 50Backstage passes to a TAFKAL80ETC concert, -4, 0Conjured Mana Cake, -6, 0